brototype.js

```diff
@@ -68,18 +68,19 @@
             if (Array.isArray(key)) {
                 var index, value, result = [];
                 for (index in key) {
-                    if (value = this.iCanHaz(key[index])) {
-                        result.push(value);
-                    }
+                    value = this.iCanHaz(key[index]);
+                    result.push(value);
                 }
                 return result;
             }
             var props = key.split('.'),
                 item = this.obj;
-            for (var i = 0; i < props.length; i++) {
-                item = item[props[i]];
-                if (Bro(item).isThatEvenAThing() === Bro.NOWAY) {
-                    return item;
+            if (item !== undefined) {
```

Comment:
Yup np you're totally right. Missed the `typeof` in your original comment. Hopefully my force push overwrote it properly - looks like it?